exporter/otlp/lib/opentelemetry/exporter/otlp/exporter.rb

```diff
@@ -341,6 +341,9 @@ def as_otlp_span_kind(kind)
 
         def as_otlp_key_value(key, value)
           Opentelemetry::Proto::Common::V1::KeyValue.new(key: key, value: as_otlp_any_value(value))
+        rescue Encoding::UndefinedConversionError => e
+          OpenTelemetry.handle_error(exception: e, message: "encoding error for key #{key} and value #{value}")
+          Opentelemetry::Proto::Common::V1::KeyValue.new(key: key, value: as_otlp_any_value('Encoding Error'))
```

Comment:
Can the key itself be malformed and cause a `Encoding::UndefinedConversionError`?

Also, is there a specification around behavior here that says `encode as value "Encoding Error"`

Reply:
>Can the key itself be malformed and cause a Encoding::UndefinedConversionError?

I'd expect it is vulnerable to the same issue.  Though we're not encoding the key with `as_otlp_any_value` as we are the value, which is why I have it this way.  

Maybe we should be? @fbogsany 

>Also, is there a specification around behavior here that says encode as value "Encoding Error"

I didn't find anything, and I'm open to alternatives/using spec'd conventions I missed.  I found some mention of discarding attributes in certain situations, but I think discarding in this situation might be more frustrating than what is present in the PR currently.  

Chasing down why attributes sometimes are not present sounds frustrating.

Reply:
The [error handling guidelines](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/error-handling.md) suggest that this is the correct thing to do: call the configurable error handler and insert a default value (there's no guidance on what that value should be in the case of invalid encoding).

While it is technically possible for the `key` to be malformed, it is unlikely to be a problem in practice because attribute keys are typically explicitly provided rather than extracted or computed from other data. I think we should deal with the known `value` problem first, so we can figure out which key & instrumentation needs to be fixed (this is likely to be an auto-instrumentation bug).